### DISCUSSION TITLE
close_vectors for Float64 vectors

### DISCUSSION THREAD

**Discussion by Lucatil** [oscar-system/Oscar.jl#4139] [Q&A] [answered]
Hello everyone,

I am relatively new to the Oscar library, and I am trying to implement something. For that, I need to find the set of closest lattice points to a given point x.
An example of what I want to do can be found in the attached file.
The error I am getting is: "MethodError: no method matching (::Nemo.QQField)(::Float64)".
Any suggestions? Am I doing something wrong? Thanks in advance :)
![Screenshot 2024-09-24 101034](https://github.com/user-attachments/assets/77dabc6e-66e9-4a59-894a-3656cbae699d)

**Comment by thofma** [comment, marked as answer]
Our support for floating point numbers is lacking (and I am not sure it will improve). It works if you supply rational numbers instead:
```
julia> x = [1//2; 1//2];

julia> CV = close_vectors(L, x, 4)
12-element Vector{Tuple{Vector{ZZRingElem}, QQFieldElem}}:
 ([1, 2], 5//2)
 ([0, 2], 5//2)
 ([2, 1], 5//2)
 ([1, 1], 1//2)
 ([0, 1], 1//2)
 ([-1, 1], 5//2)
 ([2, 0], 5//2)
 ([1, 0], 1//2)
 ([0, 0], 1//2)
 ([-1, 0], 5//2)
 ([1, -1], 5//2)
 ([0, -1], 5//2)
```

**Reply by Lucatil** [reply]
Thank you, for the quick reply. That solved the issue. :)
